test_rmw_implementation/test/test_subscription.cpp

```diff
@@ -377,3 +377,29 @@ TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), count_mismatched_subs
   EXPECT_EQ(RMW_RET_OK, ret) << rmw_get_error_string().str;
   EXPECT_EQ(0u, publisher_count);
 }
+
+TEST_F(CLASSNAME(TestSubscriptionUse, RMW_IMPLEMENTATION), take_with_bad_args) {
+  rmw_ret_t ret;
```

Comment:
Suggestion: move variable declaration to line 387 where it is used for the first time

Reply:
Updated